modules/PowermeterBSM_WS36A/main/powermeterImpl.cpp

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Qwello GmbH and Contributors to EVerest
+
+#include "powermeterImpl.hpp"
+
+#include <GenericPowermeter/serialization.hpp>
+#include <chrono>
+#include <ctime>
+#include <fmt/core.h>
+#include <stdexcept>
+#include <string>
+
+namespace {
+
+int16_t get_utc_offset() {
+    std::time_t t = std::time(nullptr);
+    const auto gmt_time = std::mktime(std::gmtime(&t));
+    const auto local_time = std::mktime(std::localtime(&t));
+
+    const auto diff_sec = std::difftime(local_time, gmt_time);
+    return static_cast<int16_t>(diff_sec / 60);
+}
+
+constexpr module::utils::Register SIGNATURE_STATUS_START{41286, 1};
+constexpr module::utils::Register SIGNATURE_STATUS_STOP{41540, 1};
+constexpr module::utils::Register SIGNATURE_START{43295, 496};
+constexpr module::utils::Register SIGNATURE_STOP{43795, 496};
+constexpr module::utils::Register UNIX_TIME{40260, 2};
+// When setting the time we have to set it with the offset - this is not clearly
+// documented in the manual.
+constexpr module::utils::Register UNIX_TIME_WITH_UTC_OFFSET{40260, 3};
+constexpr module::utils::Register UTC_OFFSET{40262, 1};
+constexpr module::utils::Register META_DATA_1{40279, 70};
+
+/// @brief The status when reading the
+constexpr uint16_t SIGNATURE_STATUS_DONE = 2;
+
+} // namespace
+
+namespace module {
+namespace main {
+
+using namespace module::utils;
+using namespace types::powermeter;
+using namespace types::serial_comm_hub_requests;
+
+void powermeterImpl::init() {
+}
+
+void powermeterImpl::ready() {
+    std::thread([this]() {
+        while (true) {
+
+            TransactionReq req;
```

Comment:
probably want to add a TODO, since this is hardly what we actually want to do in the end, right?